File: Sem4_PCC-AIML492/A02_07-04-2026/A02Q02_towerOfHanoi.py
# Implement Tower of Hanoi and count total moves.

def tower_of_hanoi(disc, source, aux, to, moves=0):
    if disc == 1:
        print(f"Move disk 1 from {source} to {to}")
        return moves + 1

    moves = tower_of_hanoi(disc - 1, source, to, aux, moves)

    print(f"Move disk {disc} from {source} to {to}")
    moves += 1

    moves = tower_of_hanoi(disc - 1, aux, source, to, moves)

    return moves

disc = int(input("Enter number of disks: "))
total_moves = tower_of_hanoi(disc, 'A', 'B', 'C')
print("Total moves:", total_moves)

File: Sem4_PCC-CS494/A02_09-04-2026/A02Q02_towerOfHanoi.c
// Implement Tower of Hanoi 

#include <stdio.h>

void towerOfHanoi(int disc, char from, char to, char aux) {
    if (disc == 1) {
        printf("%c -> %c\n", from, to);
        return;
    }

    towerOfHanoi(disc - 1, from, aux, to);
    printf("%c -> %c\n", from, to);
    towerOfHanoi(disc - 1, aux, to, from);
}

int main() {
    int disc;
    printf("Enter the number of discs: ");
    scanf("%d", &disc);

    towerOfHanoi(disc, 'A', 'C', 'B');

    return 0;
}